cdc/sinkv2/eventsink/txn/txn_sink.go

```diff
@@ -14,24 +14,197 @@
 package txn
 
 import (
+	"encoding/binary"
+	"hash/crc64"
+	"sync"
+
+	"github.com/pingcap/log"
 	"github.com/pingcap/tiflow/cdc/model"
 	"github.com/pingcap/tiflow/cdc/sinkv2/eventsink"
+	"github.com/pingcap/tiflow/pkg/causality"
+	"go.uber.org/zap"
+)
+
+const (
+	conflictDetectorSlots int64 = 1024 * 1024
 )
 
 // Assert EventSink[E event.TableEvent] implementation
 var _ eventsink.EventSink[*model.SingleTableTxn] = (*Sink)(nil)
 
 // Sink is the sink for SingleTableTxn.
-type Sink struct{}
+type Sink struct {
+	conflictDetector *causality.ConflictDetector[*worker, txnEvent]
+	workers          []*worker
+}
+
+func newSink(bes []backend) Sink {
```

Comment:
```suggestion
func newSink(backends []backend) Sink {
```
"bes" looks a little bit confusing.